src/worldlogic.rs

```diff
@@ -0,0 +1,29 @@
+use crate::world::World;
+use crate::utils::Utils;
+use crate::definitions;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+
+#[derive(Clone)]
+pub struct Worldlogic{
+    world:Rc<RefCell<World>>,
+}
+
+impl Worldlogic {
+    pub fn new(world: Rc<RefCell<World>>) -> Worldlogic {
+        Worldlogic {
+            world
+        }
+    }
+
+    pub fn _sandAction(x:i32,y:i32){
```

Comment:
Rust function names should be sname case  -> `_sand_action`

Function calls in an impl should include a self parameter, because we don't want to give ownership to self `&self` should be passed

Formatting missing whitespace should be `(x: i32, y: i32) {`